internal/receiver/discoveryreceiver/README.md

```diff
@@ -0,0 +1,334 @@
+         rule: type == "container"
+         config:
+           type: collectd/redis
+           // the metric or log statement
+           status:
+             metrics:
+               successful:
+                 - regexp: '.*'
+                   // Only emit a single log record for this status entry instead of one for each matching received metric (`false`, the default)
+                   first_only: true
+                   record:
+                     severity: info
+                     body: Successfully able to connect to Redis container.
+             statements:
+               partial:
+                 - regexp: (WRONGPASS|NOAUTH|ERR AUTH)
+                   first_only: true
+                   record:
+                     severity: warn
+                     body: Container appears to be accepting redis connections but the default auth setting is incorrect. 
+               failed:
+                 - regexp: ConnectionRefusedError
+                   first_only: true
+                   record:
+                     severity: debug
+                     body: Container appears to not be accepting redis connections.
+exporters:
+  logging:
+    logLevel: debug
+    sampling_initial: 1
+    sampling_thereafter: 1
+service:
+  extensions:
+    - docker_observer
+  pipelines:
+    logs:
+      receivers:
+        - discovery
+      exporters:
+        - logging
+```
+
+Given this configuration, if the Discovery receiver's Docker observer instance reports an active Redis container and
+the Smart Agent receiver's associated `collectd/redis` monitor is able to generate metrics for the container, something
+like the following log record would be emitted by the receiver:
+
+```
+2022-07-27T16:35:03.575Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "logging", "#logs": 1}
+2022-07-27T16:35:03.575Z	info	ResourceLog #0
+Resource SchemaURL:
+Resource labels:
+     -> container.name: STRING(lucid_goldberg)
+     -> container.image.name: STRING(redis)
+     -> discovery.receiver.rule: STRING(type == "container" && image == "redis")
+     -> discovery.endpoint.id: STRING(2f0d88c1d93b2aafce4a725de370005f9e7e961144551a3df37a88b27ebed48f:6379)
+     -> discovery.receiver.name: STRING(smartagent/redis)
+     -> event.type: STRING(metric.match)
+ScopeLogs #0
+ScopeLogs SchemaURL:
+InstrumentationScope
+LogRecord #0
+ObservedTimestamp: 2022-07-27 16:35:03.575234252 +0000 UTC
+Timestamp: 2022-07-27 16:35:01.522543616 +0000 UTC
+Severity: info
+Body: Successfully able to connect to Redis container.
+Attributes:
+     -> metric.name: STRING(counter.expired_keys)
+     -> discovery.status: STRING(successful)
+Trace ID:
+Span ID:
+Flags: 0
+```
+
+If instead the Docker observer reports an active Redis container but the `collectd/redis` authentication information is
+incorrect, the following log record would be emitted by the Discovery receiver:
+
```

Comment:
Suggestion: 

Instead, if the Docker observer reports an active Redis container but the `collectd/redis` authentication information is incorrect, the Discovery receiver will emit the following log record:

Reply:
updated